src/Language/PureScript/Pretty/JS.hs

```diff
@@ -58,9 +57,9 @@ literals = mkPattern' match'
     , return $ emit "}"
     ]
     where
-    objectPropertyToString :: (Emit gen) => Text -> gen
-    objectPropertyToString s | identNeedsEscaping s = string s
-                             | otherwise = emit s
+    objectPropertyToString :: (Emit gen) => PSString -> gen
+    objectPropertyToString s | identNeedsEscaping $ T.pack $ codePoints s = string s
```

Comment:
I am not sure that we should have any function that treats lone surrogates in that way, as it's so easy to convert them to U+FFFD by then doing Data.Text.pack on the result (and also we should be avoiding `String`). I think a `Maybe` variant would make it much easier to avoid accidentally losing information in this way.

As far as I can tell, there are no more than three things we might need to do with a PSString:
* Render it as a JS string with appropriate escapes in the JS backend
* Give it to an alternate backend in the form of UTF-16 code units
* Render it in an error message produced by the compiler

As far as I can tell, having `decodeText :: PSString -> Maybe Text` and `renderJSON` covers the first and `toUtf16CodeUnits` covers the second. For the third I think we just want to obtain a PS string literal that would parse to the same `PSString` value, which I think means we want a function that's similar to `renderJSON` except which uses PS escapes.

Given all this, and assuming I haven't missed anything, I don't see a need for this `codePoints` function.